lms/djangoapps/course_home_api/dates/v1/views.py

```diff
@@ -91,6 +92,10 @@ def get(self, request, *args, **kwargs):
             reset_masquerade_data=True,
         )
 
+        is_enrolled = CourseEnrollment.is_enrolled(request.user, course_key)
+        if not is_enrolled and not request.user.is_staff:
```

Comment:
Nit: don't need the is_enrolled variable if we aren't going to use it anywhere else. Just put it inline. Also, do you need to do the has_access call on the course rather than just the .is_staff here? not sure what subtleties could be different, but is_staff might be referring to are you a global staff user (i.e. edX) vs staff on the course